ASAN build:

```
c/nvim/CMakeFiles/nvim.dir/eval.c.o   -c ../src/nvim/eval.c
../src/nvim/eval.c:20501:5: error: implicit declaration of function 'clear_tv' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    clear_tv(current_funccal->rettv);
    ^
../src/nvim/eval.c:20501:5: note: did you mean 'clearenv'?
/usr/include/stdlib.h:595:12: note: 'clearenv' declared here
extern int clearenv (void) __THROW;
           ^
../src/nvim/eval.c:20501:5: error: this function declaration is not a prototype [-Werror,-Wstrict-prototypes]
    c/nvim/CMakeFiles/nvim.dir/eval.c.o   -c ../src/nvim/eval.c
../src/nvim/eval.c:20501:5: error: implicit declaration of function 'clear_tv' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    clear_tv(current_funccal->rettv);
    ^
../src/nvim/eval.c:20501:5: note: did you mean 'clearenv'?
/usr/include/stdlib.h:595:12: note: 'clearenv' declared here
extern int clearenv (void) __THROW;
           ^
../src/nvim/eval.c:20501:5: error: this function declaration is not a prototype [-Werror,-Wstrict-prototypes]
    clear_tv(current_funccal->rettv);
    ^(current_funccal->rettv);
    ^
```

Instead of `clear_tv` use `tv_clear`.

This was only caught by the ASAN build because it's the only build that defines `EXITFREE` (which is where `free_all_functions` is used).

--- a/src/nvim/eval.c
+++ b/src/nvim/eval.c
@@ -20498,7 +20498,7 @@ void free_all_functions(void)
 
   // Clean up the call stack.
   while (current_funccal != NULL) {
-    clear_tv(current_funccal->rettv);
+    tv_clear(current_funccal->rettv);
     cleanup_function_call(current_funccal);
   }
 

